src/compiler/crystal/semantic/call_error.cr

```diff
@@ -173,58 +103,14 @@ class Crystal::Call
       inner_exception = TypeException.for_node(similar_def, inner_msg)
     end
 
-    if owner_trace
```

Comment:
`owner_trace` was only defined in the extracted branch, so here it could only be nil. And now the compiler complains about that. See #9377
I moved the definition further down where it makes sense, because it's not used before that anyways.